src/inmanta/deploy/work.py

```diff
@@ -20,47 +20,31 @@
 import asyncio
 import functools
 import itertools
-import typing
 from collections.abc import Iterator, Mapping, Set
 from dataclasses import dataclass
-from typing import Callable, Generic, Optional, TypeAlias, TypeVar
+from typing import Callable, Generic, Optional, TypeVar
 
 from inmanta.data.model import ResourceIdStr
 from inmanta.resources import Id
 
 
 @dataclass(frozen=True, kw_only=True)
-class _Task(abc.ABC):
+class Task(abc.ABC):
```

Comment:
Why still define the ABC here? I think it would make sense to move it to `tasks` as well, so that `work` depends on `tasks` but not vice versa.

Reply:
right, that could make things better 

I found it more natural to have the abstract in one place, but I see the point